packages/algorand-js/src/interpreter/opcode-list.ts

```diff
@@ -80,3 +81,71 @@ export class Arg extends Op {
     stack.push(a);
   }
 }
+
+// load block of byte-array constants
+export class Bytecblock extends Op {
+  readonly bytecblock: Uint8Array[];
+  readonly interpreter: Interpreter;
+
+  constructor (interpreter: Interpreter, bytecblock: Uint8Array[]) {
+    super();
+    this.interpreter = interpreter;
+    this.bytecblock = bytecblock;
+  }
+
+  execute (stack: TEALStack): void {
+    this.assertArrLength(this.bytecblock);
+    this.interpreter.bytecblock = this.bytecblock;
+  }
+}
+
+// push bytes constant from bytecblock to stack by index
+export class Bytec extends Op {
+  readonly index: number;
+  readonly interpreter: Interpreter;
+
+  constructor (idx: number, interpreter: Interpreter) {
+    super();
+    this.index = idx;
+    this.interpreter = interpreter;
+  }
+
+  execute (stack: TEALStack): void {
+    const bytec = this.assertBytes(this.interpreter.bytecblock[this.index]);
```

Comment:
Please add a check that the index doesn't exceeds the array. We should have a clear message for it rather than falling down into `undefined` check.
Same for `Intc`

Reply:
Alright